### ISSUE TITLE
Implement blocked_nd_range as a fully supported feature

### ISSUE THREAD

**Issue by akukanov** [uxlfoundation/oneTBB#1539]
# Description

The oneTBB specification 1.4 will add the `blocked_nd_range` class template; the corresponding patch https://github.com/uxlfoundation/oneAPI-spec/pull/555 has been merged into the specification draft.

PR #1449 modifies the preview implementation of `blocked_rangeNd` to match the specification, including the name change, and removes the preview status.

There is an open question of supporting class template argument deduction (CTAD). Two related PRs are made on top of the current preview implementation, #1524 and #1525.

For the documentation there is an associated issue #1529 about describing multidimensional ranges in the developer guide.
# What needs to be done

The CTAD PRs mentioned above need to be updated on top of #1449 (once it is merged) and tested. The testing should include C++11 environments for the general functionality, including various ways to construct an instance without CTAD.

**Comment by kboyarinov** [comment]
`blocked_nd_range` was added as a Fully Supported Feature by d1d43ad and the explicit deduction guides were added as an experimental feature by 9d45787. 
The corresponding RFC was added into `experimental`. 
Further activities should be tracked in RFC and the in specification.
This issue can be closed.
